src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -258,15 +258,16 @@ private object[] ProcessInputParameters(object[] parameters)
                 using (_metricsLogger.LatencyEvent(eventName))
                 {
                     ICompilation compilation = _compilationService.GetFunctionCompilation(Metadata);
+
+                    Assembly assembly = compilation.EmitAndLoad(cancellationToken);
```

Comment:
With this change, we now emit before validation. This does remove an optimization in the case of C# where we can perform code analysis for the small check we run below, but it's not significant in this code path (this is not triggered as part of diagnostics compilations) and the complexity to handle both scenarios (where emit is required and when it isn't) is probably not justified at this point.